### PR TITLE
Cleanup before 1.2 release

### PR DESCRIPTION
**Pull request by jeremiah-k**
## Summary by CodeRabbit

- Documentation
  - Revamped E2EE and authentication guidance: standardizes recommended login, adds MAS/OIDC cautions, marks manual tokens as legacy with expiry warnings, updates Windows guidance, and adds “What’s New” 1.2 notes.

- New Features
  - Added a documented logout command for secure session cleanup.
- Chores
  - Weather handling now emits a fallback warning when a timestamp anchor is missing (behavior unchanged).
- Tests
  - Expanded tests for weather edge cases, exception handling, and broadcast message detection.

### CODE REVIEW

**Comment by jeremiah-k** [comment]
@coderabbitai review

File: README.md [hunk 1]
```diff
@@ -29,29 +29,6 @@ Visit our [Wiki](https://github.com/jeremiah-k/meshtastic-matrix-relay/wiki) for
 
 ---
```
[the remark this answers is not in the document]
Comment (jeremiah-k) [reply README.md]: @gemini-code-assist There are too many deployment options and there is no "best" way to get started, users should read the INSTRUCTIONS.md to get started,  README.md is now an intro/overview of the project.  No need to flag this in the future.

**Comment by jeremiah-k** [comment]
@coderabbitai review
